lib/fs/util_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2019 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package fs
+
+import (
+	"runtime"
+	"testing"
+)
+
+func TestCommonPrefix(t *testing.T) {
+	test := func(first, second, expect string) {
+		t.Helper()
+		res := CommonPrefix(first, second)
+		if res != expect {
+			t.Errorf("Expected %s got %s", expect, res)
+		}
+	}
+
+	if runtime.GOOS == "windows" {
+		test(`c:\Audrius\Downloads`, `c:\Audrius\Docs`, `c:\Audrius`)
+		test(`c:\Audrius\Downloads`, `C:\Audrius\Docs`, ``) // Case differences :(
```

Comment:
I vaguely remember us handling drive letter case somehow in `basicFilesystem`, but not sure - more a note to self than an actionable review item.